### PR TITLE
Add MSYS2 action

### PR DESCRIPTION
**Pull request by hyoklee**
Some tests fail on all 4 environments.
However, compilation works fine.
clang17 environment needs extra definitions.

### CODE REVIEW

File: .github/workflows/msys2.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            ninja:p
+      - name: Test HDF5
+        shell: msys2 {0}
+        run: |
+          uname -a
+          git clone https://github.com/HDFgroup/hdf5.git
+          cd hdf5
+          mkdir build
+          cd build
+          cmake -DH5_HAVE_ALARM:INTERNAL=0 -DH5_HAVE_ASPRINTF:INTERNAL=0 -DH5_HAVE_VASPRINTF:INTERNAL=0 ..
```
Comment (derobins) [inline .github/workflows/msys2.yml:55]: We should probably move the alarm, asprintf, and vasprintf checks in CMake so they check on Windows (I'm assuming they are not checked on Win32).

Comment (derobins) [reply .github/workflows/msys2.yml:55]: Interestingly, they are run all the time. We'll need to figure out why they are not being detected on Windows w/ MSYS2.

**Review by derobins — APPROVED** [review]

File: .github/workflows/msys2.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            cmake:p
+            ninja:p
+      - name: Test HDF5
+        shell: msys2 {0}
+        run: |
+          uname -a
+          git clone https://github.com/HDFgroup/hdf5.git
+          cd hdf5
+          mkdir build
+          cd build
```
Comment (byrnHDF) [inline .github/workflows/msys2.yml:54]: Please add steps between configure, build and test

File: .github/workflows/msys2.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
+name: msys2
```
Comment (byrnHDF) [inline .github/workflows/msys2.yml:1]: Change name to match other yml files and add job name with matrix.sys variable

File: .github/workflows/msys2.yml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+on:
+  workflow_dispatch:
+  push:
```
Comment (byrnHDF) [inline .github/workflows/msys2.yml:5]: At only 12 mins of time we should make this a callable workflow from the cmake.yml.

File: .github/workflows/msys2.yml (hunk 1 above)
Comment (byrnHDF) [inline .github/workflows/msys2.yml:55]: Is there is a reason to not use the same configure / build / test steps we use in other CMake yml files?

**Comment by byrnHDF** [comment]
We should fix the issues raised here and then create a solution that addresses all the issues for MSYS testing.

**Comment by byrnHDF** [comment]
Superseded by #3938

**Pull request closed without merging**